client/hotrod-client/src/test/java/org/infinispan/client/hotrod/counter/RemoteCounterManagerTest.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.client.hotrod.counter;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.infinispan.client.hotrod.RemoteCounterManagerFactory;
+import org.infinispan.commons.util.Util;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.counter.api.CounterManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.server.hotrod.counter.CounterManagerTest;
+import org.infinispan.server.hotrod.counter.impl.CounterManagerImplTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.util.logging.Log;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * A {@link CounterManager} implementation test.
+ *
+ * @author Pedro Ruivo
+ * @since 9.2
+ */
+@Test(groups = "functional", testName = "client.hotrod.counter.RemoteCounterManagerTest")
+public class RemoteCounterManagerTest extends AbstractCounterTest implements CounterManagerTest {
+
+   private static final String PERSISTENT_LOCATION = TestingUtil.tmpDirectory("RemoteCounterManagerTest");
+   private static final String TMP_LOCATION = PERSISTENT_LOCATION + File.separator + "tmp";
+   private final CounterManagerTest testImpl;
+
+
+   public RemoteCounterManagerTest() {
+      testImpl = new CounterManagerImplTest(this::allTestCounterManagers, this::log, this::cacheManager);
```

Comment:
fyi, you can use `@Factory` method to replace the actually tested instance (its methods have to be annotated with `@Test` ofc)

Reply:
IMO, this is simpler instead of trying to adapt it to use the `@Factory`

Reply:
@pruivo I don't like the idea of having classes that end in `Test` but aren't really tests. A test strategy maybe?

+1 to use `@Factory` or even inheritance, we shouldn't bend ourselves backwards just because TestNG is weird about how groups are inherited.

Reply:
If I use the `@Factory` annotation, it fails
```
java.lang.IllegalStateException: Test org.infinispan.client.hotrod.counter.WeakCounterAPITest extends MultipleCacheManagersTest and declares its own @Factory method: 
```
If I extend the `factory()` method as 
```
   @Override
   public Object[] factory() {
      return new Object[]{
            new WeakCounterImplTest(),
            new WeakCounterAPITest()
      };
   }
```
the tests aren't invoked and they don't run.

Reply:
@danberindei AFAIK, I can't extend 2 abstract classes :)

Reply:
@pruivo That's easy to fix, don't extend `AbstractCounterTest` :)
Although I never liked extending tests from one module in 

Re: `@Factory`, yes, you're not supposed to use it in subclasses of `MultipleCacheManagersTest`. We could probably allow it, only I'm not 100% sure how TestNG handles multiple `@Factory` annotations so I'd rather make all the tests use `@Factory` and skip the tests explicitly instead of using the `@InCacheMode` and similar annotations

Also, I recently (ISPN-8537) changed `factory()` to check for copy-paste errors and require the returned objects to be instances of the concrete test class. Unfortunately I lost the `@Test` annotation in one of my iterations and now "wrong" instance types are ignored instead of failing explicitly. I will have a PR soon.

Reply:
-1 I don't want to copy the `MultiHotRodServersTest` 3 times... (I'm running 3 tests from the server module)
I'll change the name to test strategy as suggested above